tests/unit/test_charm.py

```diff
@@ -3395,10 +3417,180 @@ def test_given_certificate_rejected_in_notary_when_configure_then_certificate_re
                     "is_initialized.return_value": True,
                     "token_is_valid.return_value": True,
                     "get_certificate_requests_table.return_value": CertificateRequests(
-                        rows=[CertificateRequest(id=1, csr=str(csr), certificate_chain="rejected")]
+                        rows=[
+                            CertificateRequestRow(id=1, csr=str(csr), certificate_chain="rejected")
+                        ]
                     ),
                 },
             ),
         ):
             context.run(context.on.update_status(), state)
         mock_set_relation_certificate.assert_called_once()
+
+    @patch(f"{TLS_LIB_PATH}.TLSCertificatesRequiresV4.get_assigned_certificate")
+    def test_given_access_relation_created_when_configure_then_certificate_not_replaced(
+        self, mock_assigned_certificates, context
+    ):
+        with tempfile.TemporaryDirectory() as tempdir:
```

Comment:
I believe it is preferable to use pytest's [`tmp_path`](https://docs.pytest.org/en/stable/how-to/tmp_path.html).

Reply:
done